### PR TITLE
Add register to users table

### PR DESCRIPTION
**Pull request by dankmitchell**
### Context
No need to separate `users` and `download_users` in the database.

### Changes proposed in this pull request
- Add `register` attr to `users` table
- Migrate all date from `download_users` into `users`
- Update `download_users` controller to save to `users` table
- Fix specs

### Guidance to review
Running the migrations should copy the data from `download_users` into `users` and completing the download form and api key form on the frontend should results in the data being saved to the `users` table

### CODE REVIEW

**Comment by gidsg** [comment]
would be nice to add a `user_type` column to make it easier to distinguish the users. This could be an enum (either Postgres or Rails int style).

**Comment by michaelabenyohai** [comment]
The migration doesn't populate user_type for existing download or API users

File: db/migrate/20180612082738_move_download_users_to_users.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  def change
+    DownloadUser.find_each do |download_user|
+      User.create(
+        email: download_user.email,
+        register: download_user.register,
+        non_gov_use_category: download_user.non_gov_use_category,
+        department: download_user.department,
+        is_government: download_user.is_government,
+        api_key: nil,
+        contactable: false
```
Comment (michaelabenyohai) [inline db/migrate/20180612082738_move_download_users_to_users.rb]: Just curious why contactable is false for existing download users?

Comment (michaelabenyohai) [reply db/migrate/20180612082738_move_download_users_to_users.rb]: Does this also need to be populated for API users?

**Comment by michaelabenyohai** [comment]
When giving my details after download it doesn't appear to populate the `non_gov_use_category` or `department` in the `users` table.

**Comment by michaelabenyohai** [comment]
When trying to get an API key registers-selfservice returns a 500 error. Does anyone else get this?

**Comment by gidsg** [comment]
@michaelabenyohai have you definitely run migrations on self-service. If so, what is the exception in the self-service logs?

**Comment by gidsg** [comment]
@michaelabenyohai ... thinking more about it, you can get a 500 if you don't have`GOOGLE_CREDENTIALS_JSON` set.

**Comment by michaelabenyohai** [comment]
Yep that was it @gidsg! Sorry forgot about that.

When I choose not government we don't populate the `contactable` column for API key users. I assume this is because we don't ask them. Is it worth us populating this with false? What does it mean if this column is nil?

**Comment by gidsg** [comment]
@michaelabenyohai 
> What does it mean if this column is nil?

currently we would interpret it as `false` this is because it would apply to early users created prior to this field existing. We could clarify it in a subsequent migration though.

File: db/migrate/20180612082738_move_download_users_to_users.rb [hunk 2]
```diff
@@ -8,7 +8,7 @@ def change
         department: download_user.department,
         is_government: download_user.is_government,
         api_key: nil,
-        contactable: false
+        contactable: download_user.created_at.between?(Time.utc(2018, 0o5, 30, 18), Time.now)
```
Comment (dankmitchell) [inline db/migrate/20180612082738_move_download_users_to_users.rb]: Is this a typo @gidsg ? Should be `Time.utc(2018, 05, 30, 18)`

Comment (gidsg) [reply db/migrate/20180612082738_move_download_users_to_users.rb]: @dankmitchell linter made me do it m8 https://github.com/bbatsov/ruby-style-guide#numeric-literal-prefixes

**Review by michaelabenyohai — APPROVED** [review]

**Pull request merged**